modules/core/include/opencv2/core/mat.hpp

```diff
@@ -1084,6 +1084,18 @@ class CV_EXPORTS Mat
     */
     void copyTo( OutputArray m, InputArray mask ) const;
 
+#if CV_FP16
```

Comment:
1) please, put convertFp16 out of cv::Mat. I'm fine to have it as a separate function.
2) such API cannot be put under conditional compilation; if we add a function to convert to/from fp16, we should always provide it, i.e. there should be software emulation of such function if HW cannot do that